### PR TITLE
Add device based MQTT auto discovery

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add device based MQTT auto discovery.

This is de docs PR for the reworked and reverted PR https://github.com/home-assistant/core/pull/109030

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/118757
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Updated MQTT integration documentation to include a new required field `platform` for each integration.
	- Clarified the `unique_id` field's requirements for device-based discovery.
	- Introduced a new section on **Device discovery payload** detailing the structure and requirements for device discovery.
	- Added a section on **Discovery messages** with comprehensive explanations of the topic format and payload structure.
	- Expanded the section on availability topics to explain device announcements and retained messages.
	- Added a new section on **Adding information about the origin of a discovery message**.
	- Provided **Discovery examples with component discovery** for setting up various MQTT components.
	- Enhanced clarity on the naming of MQTT entities, explaining how `entity_id` is generated.
	- Updated documentation to emphasize the necessity of unique identifiers to avoid duplicate entries during device reconnections.

### CODE REVIEW

File: source/_integrations/humidifier.mqtt.markdown [hunk 1]
```diff
@@ -266,6 +266,10 @@ modes:
   required: false
   type: [list]
   default: []
+platform:
+  description: Must be `humidifier`. Only allowed and required in [MQTT auto discovery device messages](/integrations/mqtt/#device-discovery-payload).
+  required: true
+  type: string
```
[the remark this answers is not in the document]
Comment (jbouwh) [reply source/_integrations/humidifier.mqtt.markdown:272]: Not sure what this suggestion changes

[the remark this answers is not in the document]
Comment (jbouwh) [reply source/_integrations/humidifier.mqtt.markdown:272]: Okay, but it is already fixed IMO.

**Review by c0ffeeca7 — APPROVED** [review]
Hej, you didn't leave anything for me to nit-pick on. :laughing: 
Thank you, @jbouwh :+1:

**Comment by jbouwh** [comment]
> Hej, you didn't leave anything for me to nit-pick on. 😆 Thank you, @jbouwh 👍

I guess @coderrabbitai challenged me, and caused some concurrency. Thanks 😊 👍

**Pull request merged**